appwindow.js

```diff
@@ -107,6 +107,40 @@ var AppWindow = GObject.registerClass(
             this.method_handler(this.settings, 'changed::window-position', this.update_resize_boxes);
             this.update_resize_boxes();
 
+            this.simple_action('window-add-height', () => {
+                const current = this.settings.get_double('window-size');
+
+                if (current > 0.95)
+                    this.settings.set_double('window-size', 1.0);
+                else
+                    this.settings.set_double('window-size', current + 0.05);
+            });
+            this.simple_action('window-sub-height', () => {
+                const current = this.settings.get_double('window-size');
+
+                if (current < 0.05)
+                    this.hide();
+                else
+                    this.settings.set_double('window-size', current - 0.05);
+            });
+
+            this.simple_action('window-add-opacity', () => {
+                const current = this.settings.get_double('background-opacity');
+
+                if (current > 0.95)
+                    this.settings.set_double('background-opacity', 1.0);
+                else
+                    this.settings.set_double('background-opacity', current + 0.05);
+            });
+            this.simple_action('window-sub-opacity', () => {
+                const current = this.settings.get_double('background-opacity');
+
+                if (current < 0.05)
+                    this.settings.set_double('background-opacity', 0.0);
+                else
+                    this.settings.set_double('background-opacity', current - 0.05);
```

Comment:
Could you extract this into a function? `adjust_double_setting(name, difference)`. And then use it to implement all 4 actions

Reply:
Yes, I think it also might make sense to add optional `max` and `min` parameters as well to this.

However, I am running into an issue with this:
`this.simple_action('window-inc-opacity', this.adjust_double_setting('background-opacity', opacity_mod));`

As the `signal_connect` seems to want a function - I can just wrap it like:
`this.simple_action('window-inc-opacity', () => {this.adjust_double_setting('background-opacity', opacity_mod)});`

But there is likely a better solution...

Reply:
> As the `signal_connect` seems to want a function - I can just wrap it like:
> `this.simple_action('window-inc-opacity', () => {this.adjust_double_setting('background-opacity', opacity_mod)});`

Totally acceptable solution (`opacity_mod` is a constant, I guess? If so, please use `UPPER_CASE`).

> Yes, I think it also might make sense to add optional `max` and `min` parameters as well to this.

If I'm not mistaken, min=0 and max=1 for both settings, so these parameters aren't currently necessary. If/when they become necessary, they could be extracted from the settings schema.

Reply:
Ok great, and the min and max are necessary for the assignment (don't want to increase/substract past range).

How would we extract these props from the schema?

Reply:
https://gjs-docs.gnome.org/gio20~2.66p/gio.settings#property-settings_schema
https://gjs-docs.gnome.org/gio20~2.66p/gio.settingsschema#method-get_key
https://gjs-docs.gnome.org/gio20~2.66p/gio.settingsschemakey#method-get_range

However, the docs say that:
> This information should not be used by normal programs. It is
> considered to be a hint for introspection purposes. Normal programs
> should already know what is permitted by their own schema. The
> format may change in any way in the future -- but particularly, new
> forms may be added to the possibilities described above.

So maybe it's not a good idea